app/controllers/concerns/vaccination_mailer_concern.rb

```diff
@@ -48,6 +48,60 @@ def send_vaccination_deletion(vaccination_record)
     end
   end
 
+  def send_vaccination_discovered_if_required(vaccination_record)
+    return if vaccination_record.sourced_from_service?
+    if VaccinatedCriteria.call(
+         programme: vaccination_record.programme,
+         academic_year: AcademicYear.current,
+         patient: vaccination_record.patient,
+         vaccination_records:
+           vaccination_record.patient.vaccination_records.where.not(
+             id: vaccination_record.id
+           )
+       )
+      return
+    end
+
+    consents = vaccination_record.patient.consents
+
+    consents =
+      consents.where(
+        "vaccination_record_discovered_sent_at < ?",
+        vaccination_record.created_at
+      ).or(consents.where("vaccination_record_discovered_sent_at IS NULL"))
+
+    consents =
+      ConsentGrouper.call(
+        consents,
+        programme_id: vaccination_record.programme_id,
+        academic_year: vaccination_record.academic_year
+      )
+
+    consents = consents.select(&:response_given?).reject(&:withdrawn_at)
+
+    parents = consents.map(&:parent).uniq
```

Comment:
I think we should try and avoid the duplicate logic between here and `parents_for_vaccination_mailer`, and it might be worth refactoring `parents_for_vaccination_mailer` in to two methods, one which takes the vaccination record and another which gets the parents from a list of consents.